File: 1548-CheckIfAll1sAreAtLeastLengthKPlacesAway/1548-CheckIfAll1sAreAtLeastLengthKPlacesAway.py
# Last updated: 2/10/2026, 8:47:39 PM
class Solution(object):
    def kLengthApart(self, nums, k):
        n1 = 0
        n = 0
        l = 0

        for i in nums:
            if i == 0:
                n += 1
            if i == 1:
                if n1 > 0 and n < k:   
                    return False
                n1 += 1
                n = 0                  

        return True

File: 1548-CheckIfAll1sAreAtLeastLengthKPlacesAway/1548-CheckIfAll1sAreAtLeastLengthKPlacesAway.java
// Last updated: 8/8/2026, 9:51:09 PM
class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        final int n=nums.length;
        int last=-100000000;
        for(int i=0; i<n; i++){
            if (nums[i]==1){
                if (i-last-1<k) return false;
                last=i;
            }
        }
        return true;
        
    }
}